src/main/js/controls/Bubble/helpers/helpers.js

```diff
@@ -0,0 +1,478 @@
+            .attr("transform", transformPoint(scale)(value))
+            .attr("aria-disabled", utils.isDefined(value.onClick))
+            .attr("aria-hidden", true)
+            .attr("aria-describedby", value.key)
+            .append("circle")
+            .attr(
+                "r",
+                (d) =>
+                    decideRadius(dataTarget, d) +
+                    constants.DEFAULT_BUBBLE_SELECTOR_RADIUS
+            );
+    };
+    pointGroupPath
+        .append("g")
+        .classed(styles.pointGroup, true)
+        .each(function(d, i) {
+            const dataPointSVG = d3.select(this);
+            renderSelectionPath(dataPointSVG, d, i);
+            renderDataPoint(dataPointSVG, d, i);
+        });
+};
+
+/**
+ * Checks if the weight object is defined with min and max values.
+ *
+ * @private
+ * @param {object} dataTarget - data for the bubble graph
+ * @returns {boolean} - returns true if weight is defined and inside weight min and max is also defined else false.
+ */
+const areWeightsDefined = (dataTarget) =>
+    utils.isDefined(dataTarget.weight)
+        ? utils.isDefined(dataTarget.weight.min) &&
+          utils.isDefined(dataTarget.weight.max)
+        : false;
+
+/**
+ * Checks if hue is defined in the input JSON to get color gradient.
+ *
+ * @private
+ * @param {object} hue - hue is object defining the color range.
+ * @returns {boolean} - returns true if hue is defined else false.
+ */
+const isHueDefined = (hue) => utils.isDefined(hue);
+
+/**
+ * Decides the radius for each bubble
+ *
+ * @private
+ * @param {object} dataTarget - data for the bubble graph
+ * @param {number} value - data point whose radius has to be decided
+ * @returns {number} - returns the radius of the bubble
+ */
+const decideRadius = (dataTarget, value) => {
+    if (
+        areWeightsDefined(dataTarget) &&
+        utils.isUndefined(dataTarget.weight.maxRadius)
+    ) {
+        return bubbleScale(dataTarget)(value.weight);
+    } else if (utils.isUndefined(dataTarget.weight)) {
+        return constants.DEFAULT_BUBBLE_RADIUS_MAX;
+    } else {
+        return dataTarget.weight.maxRadius;
+    }
+};
+
+/**
+ * Decides the color for each bubble
+ *
+ * @private
+ * @param {object} dataTarget - data for the bubble graph
+ * @param {number} value - data point whose color has to be decided
+ * @returns {string} - returns color string for each bubble
+ */
+const decideColor = (dataTarget, value) => {
+    if (isHueDefined(dataTarget.hue) && areWeightsDefined(dataTarget)) {
+        return generateColor(dataTarget)(bubbleScale(dataTarget)(value.weight));
+    } else if (
+        isHueDefined(dataTarget.hue) &&
+        areWeightsDefined(dataTarget) === false
+    ) {
+        return generateColor(dataTarget)(value.y);
+    } else {
+        return dataTarget.color;
+    }
+};
+/**
+ * Handler for Request animation frame, executes on resize.
+ *  * Order of execution
+ *      * Redraws the content
```

Comment:
This is not applicable anymore. So might want to remove

Reply:
This is still getting used so I will prefer to keep it.

Reply:
You removed it from line https://github.com/cerner/carbon-graphs/pull/145/files#diff-c702c7e611aff99b05c77fd717866757R372, so the JSDoc is not accurate.

Reply:
This I am not getting still what change I need to do, can you please explain a little.

Reply:
It seems like you have incorrectly copied the JSDoc. Previously [here](https://github.com/cerner/carbon-graphs/blob/master/src/main/js/controls/Line/helpers/helpers.js#L448) we used to call `redraw()` and then `processRegions(config, canvasSVG);` but now we are only calling `processRegions(config, canvasSVG);`. So the JSDoc no longer makes sense to say `Redraws the content`

Reply:
OK got it, I will change the JSDoc for this.